src/inputParser.ts

```diff
@@ -16,10 +16,10 @@ export function parse(contents: string, stage?: string): InputDocument {
     const value = document[key];
     if (typeof value === 'string' || typeof value === 'number' || typeof value === 'boolean') {
       result[key] = { value: value };
-    } else if (value.optional && value.value) {
+    } else if (value.optional && typeof value.value !== 'undefined') {
```

Comment:
Any reason for using typeof here instead of just checking for undefined?  Any issue with swapping that out?

Reply:
It’s the safest way to do this check. `undefined` can actually be reassigned so this has been established at the best practice.

https://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined

Happy to change if you want though.

Reply:
God I hate JavaScript sometimes lol